src/main/java/me/jellysquid/mods/sodium/client/render/chunk/compile/ChunkBuilder.java

```diff
@@ -50,22 +52,85 @@ public class ChunkBuilder<T extends ChunkGraphicsState> {
     private World world;
     private BlockRenderPassManager renderPassManager;
 
-    private final int limitThreads;
+    // This is the initial number of threads we spin up upon ChunkBuilder creation.
+    // Default: 2-4 depending on render distance. *Always* less than hardLimitThreads.
+    private final int initialThreads;
+    // This is the number of threads we want to 'quickly' get up to. This is calculated
+    // by getOptimalThreadCount, but can also be configured by the user. Always less than hardLimitThreads.
+    private final int targetThreads;
+    // This is the number of threads we are allowed to create in total.
+    // This is defaulted to targetThreads, but maxes out at 64. Testing would be required to determine what
+    // actual count is optimal for a specific user and use case.
+    private final int hardLimitThreads;
+    // This is the initial time when this builder is created. We use this to create more threads.
+    private long lastThreadAddition;
+    // This is the user-configurable time delta to create a new thread, up until targetThreads.
+    private final long quickThreadCreationInterval;
+    // This is the user-configurable time delta to create a new thread, up until hardLimitThreads.
+    private final long slowThreadCreationInterval;
+
     private final ChunkVertexType vertexType;
     private final ChunkRenderBackend<T> backend;
 
     public ChunkBuilder(ChunkVertexType vertexType, ChunkRenderBackend<T> backend) {
         this.vertexType = vertexType;
         this.backend = backend;
-        this.limitThreads = getOptimalThreadCount();
+
+        // User-configurable options for chunk threads.
+        int desiredTargetThreads = SodiumClientMod.options().advanced.targetChunkThreads;
+        int desiredInitialThreads = SodiumClientMod.options().advanced.initialChunkThreads;
+
+        // These are bounded by the options configuration. Both are measured in milliseconds.
+        this.quickThreadCreationInterval = SodiumClientMod.options().advanced.quickThreadCreationInterval;
+        this.slowThreadCreationInterval = SodiumClientMod.options().advanced.slowThreadCreationInterval;
+
+        // Our hard limit of threads. Cap at 64, prefer desiredMaxThreads, otherwise use optimal thread count.
+        this.hardLimitThreads = getMaxThreadCount();
+        // Our targeted number of threads.
+        this.targetThreads = Math.min(desiredTargetThreads == 0 ? getDefaultTargetThreads() : desiredTargetThreads, this.hardLimitThreads);
+        // Our initial threads. A bit of a silly calculation for this one.
+        this.initialThreads = Math.min(desiredInitialThreads == 0 ? (SodiumWorldRenderer.getInstance().getRenderDistance() / 10) + 2 : desiredInitialThreads, this.targetThreads);
```

Comment:
Not really important but this might look a bit more readable if all these calculations were formatted the same, for example:
```
        // Our hard limit of threads. Cap at 64, prefer desiredMaxThreads, otherwise use optimal thread count.
        this.hardLimitThreads = desiredMaxThreads == 0 ? getLogicalCoreCount() : desiredMaxThreads;
        // Our targeted number of threads.
        this.targetThreads = Math.min(desiredTargetThreads == 0 ? getDefaultTargetThreads() : desiredTargetThreads, this.hardLimitThreads);
        // Our initial threads. A bit of a silly calculation for this one.
        this.initialThreads = Math.min(desiredInitialThreads == 0 ? getDefaultInitialThreads() : desiredInitialThreads, this.targetThreads);
```
Also `hardLimitThreads` technically isnt capped at 64 if someone has a CPU with more than 64 cores

Reply:
`getMaxThreadCount` is split out so that SeedQueue can mix in to it - I don't think we can inline it here.

> Also hardLimitThreads technically isnt capped at 64 if someone has a CPU with more than 64 cores

Good point...

Also, cleaned up the initial thread initializer.

Reply:
SeedQueues injector could be changed to target the field set with a WrapOperation, but this works too so I don't see an issue with it

Reply:
Makes sense, I'm not super comfortable with mixins which is why I did it this way, as long as it's not too awkward 👍